cornac/models/dnntsp/dnntsp.py

```diff
@@ -0,0 +1,652 @@
+import itertools
+import random
+from collections import defaultdict
+from typing import List
+
+import dgl
+import dgl.function as fn
+import numpy as np
+import torch
+import torch.nn as nn
+from tqdm.auto import trange
+
+from ..mf.backend_pt import OPTIMIZER_DICT
```

Comment:
I would recommend avoid importing modules from other models and make the current model self-contained. This will help us later when restructuring the models folder.